devel/delete-from-ghcr.js

```diff
@@ -98,20 +98,13 @@ module.exports = async ({fetch, octokit, tag, token}) => {
 
       if (deleteVersionError.response.data.message == "You cannot delete the last tagged version of a package. You must delete the package instead.") {
 
-        console.log(`Deleting the package ${org}/${packageName} ...`);
-        try {
-          await octokit.request('DELETE /orgs/{org}/packages/{package_type}/{package_name}', {
-            org: org,
-            package_type: 'container',
-            package_name: packageName,
-          });
-          console.log("Done.");
-        } catch (deletePackageError) {
-          console.log(deletePackageError);
-          console.error(`Could not delete ${org}/${packageName}.`);
-          errorEncountered = true;
-          continue;
-        }
+        // The right thing to do would be to delete the package
+        // ${org}/${packageName}. However, this is a potential cause for
+        // transient 403 errors¹ on GitHub Actions, so we'll keep one tagged
+        // version around as a workaround until the underlying issue is fixed.
+        // ¹ https://github.com/nextstrain/docker-base/issues/131
+        console.log(`Not deleting ${org}/${packageName}:${tag} since that requires deleting the package.`);
+
```

Comment:
_non-blocking_

Hm, kinda hacky, huh?

This makes me want to reconsider using a different registry for CI. Ideally we'd be able to use something like a workflow-local `registry` service, similar to what we do in local dev. GitHub Actions [service containers](https://docs.github.com/en/actions/using-containerized-services/about-service-containers) are job-scoped though not workflow-scoped, so we can't just use those as-is. We could, however, if we configured shared temporary storage (e.g. on S3). Alternatively, we could use some other private container registry for CI, like ECR.

Either way, such a change would also be motivated by the expectation of eventually wanting to distribute our released images via ghcr.io in addition to or instead of docker.io (due to the latter's usage limits, ToS, separate users/auth, etc).

Reply:
I rewrote this into an issue: #164